cmd/collector/app/builder/span_handler_builder.go

```diff
@@ -48,157 +45,113 @@ var (
 	errMissingElasticSearchConfig = errors.New("ElasticSearch not configured")
 )
 
-// SpanHandlerBuilder builds span (Jaeger and zipkin) handlers
-type SpanHandlerBuilder interface {
-	BuildHandlers() (app.ZipkinSpansHandler, app.JaegerBatchesHandler, error)
+// SpanHandlerBuilder holds configuration required for handlers
+type SpanHandlerBuilder struct {
+	logger         *zap.Logger
+	metricsFactory metrics.Factory
+	collectorOpts  *CollectorOptions
+	storageBuilder storageBuilder
```

Comment:
Right now you create both C and ES span writers during `NewSpanHandlerBuilder`, and wrap them into `storageBuilder` function that _looks like_ it's creating the writer but in practice just returns already created writer from a closure-captured reference. Unless you see some specific need for caring when exactly the writer is created, I think replacing the `storageBuilder` with just span writer will simplify the code.